tracer/src/Datadog.Trace/Debugger/Symbols/SymbolPdbExtractor.cs

```diff
@@ -0,0 +1,250 @@
+            methodName = MetadataReader.GetString(method.Name);
+            if (!methodName.Equals(notGeneratedMethodName.ToString()))
+            {
+                return null;
+            }
+        }
+        else if (method.Handle.RowId == cdi.StateMachineKickoffMethodRid)
+        {
+            var kickoffDef = MetadataReader.GetMethodDefinition(MethodDefinitionHandle.FromRowId(cdi.StateMachineKickoffMethodRid));
+            methodName = MetadataReader.GetString(kickoffDef.Name);
+        }
+        else
+        {
+            return null;
+        }
+
+        var closureMethodScope = CreateMethodScope(nestedType, generatedMethod);
+        closureMethodScope.Name = methodName;
+        closureMethodScope.ScopeType = SymbolType.Closure;
+        return closureMethodScope;
+    }
+
+    private Symbol[]? GetLocalSymbols(int rowId, List<DatadogMetadataReader.DatadogSequencePoint> sequencePoints)
+    {
+        List<Symbol>? localsSymbol = null;
+        var generatedClassPrefix = Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(GeneratedClassPrefix);
+
+        if (DatadogMetadataReader.GetAsyncAndClosureCustomDebugInfo(rowId).StateMachineHoistedLocal && DatadogMetadataReader.IsCompilerGeneratedAttributeDefinedOnType(MetadataReader.GetMethodDefinition(MethodDefinitionHandle.FromRowId(rowId)).GetDeclaringType().RowId))
+        {
+            localsSymbol = new List<Symbol>();
+            var fields = MetadataReader.GetTypeDefinition(MetadataReader.GetMethodDefinition(MethodDefinitionHandle.FromRowId(rowId)).GetDeclaringType()).GetFields();
+            foreach (var fieldHandle in fields)
+            {
+                var field = MetadataReader.GetFieldDefinition(fieldHandle);
+                var fieldName = MetadataReader.GetString(field.Name);
+                var span = Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(fieldName);
+                if (Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.IndexOf(span, generatedClassPrefix, StringComparison.Ordinal) == 0)
+                {
+                    continue;
+                }
+
+                var localName = Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(MetadataReader.GetString(field.Name));
+                if (localName[0] == '<')
+                {
+                    var endNameIndex = localName.IndexOf('>');
+                    if (endNameIndex > 1)
+                    {
+                        localName = localName.Slice(1, endNameIndex - 1);
+                    }
+                }
+
+                localsSymbol.Add(new Symbol
+                {
+                    Name = localName.ToString(),
+                    Type = field.DecodeSignature(new TypeProvider(false), 0),
+                    SymbolType = SymbolType.Local,
+                    Line = UnknownLocalLine // todo: get the correct line
```

Comment:
nitpick: are you able to fix that in this PR? I don't want it be abandoned in the oblivion of todo comments.

Reply:
Let's take it offline please and I'll update this comment later on.

Reply:
Okay. Let's keep this conversation on so we'll be able to come back to it later.

Reply:
We agreed to put 0 for these locals. 0c6963078c2c2ad7f8e3fa0fed81afa8ff22b82e